### ISSUE TITLE
`Serial` violates the "transitivity" contract of `PartialOrd`

### ISSUE THREAD

**Issue by kennytm** [timwie/serial-num#1]
The [PartialOrd trait](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html) requires `<` to be transitive, i.e.

```rust
if a < b && b < c {
    assert!(a < c);
}
```

This cannot be satisfied by the comparison defined by RFC1982 §3.2, e.g.

```rust
#!/usr/bin/env -S cargo +nightly -Zscript
---
[dependencies]
serial-num = "0.8.0"
---

fn main() {
    let a = serial_num::Serial::default();
    let b = a + 21845;
    let c = b + 21845;
    assert_eq!(a < b, true);
    assert_eq!(b < c, true);
    assert_eq!(a < c, false); //  showing transitivity failed
}
```

Unfortunately I think the `PartialOrd` impl needs to be removed and `cmp()` should be made an inherent method.

**Comment by timwie** [comment]
Yeah, that's a good catch. No operators for `SerialNum` I guess :-(

Might want to add `lt`, `le`, `gt`, and `ge` methods as well to use in place of `<`, `<=`, `>`, and `>=`.

**Comment by timwie** [comment]
I've named the methods `precedes()`, `precedes_or_eq()`, `succeeds()`, and `succeeds_or_eq()` instead.

**Comment by timwie** [comment]
Released as part of 0.9.0
